duplicate symbol '__lcompilers_abs_f64_' when trying to generate a shared library using clang

> a.f90

```fortran
subroutine a()
double precision :: x
x = -1.0D0
x = abs(x)
end subroutine
```

> b.f90

```fortran
subroutine b()
double precision :: x, y
x = -1.0D0
y = abs(x)
end subroutine
```


```console
% lfortran a.f90 -o a1.o -c --all-mangling --mangle-underscore --generate-object-code
% lfortran b.f90 -o a2.o -c --rtlib --all-mangling --mangle-underscore --generate-object-code
% clang -shared -fPIC -o lib.so a1.o a2.o -L/Users/pranavchiku/repos/lfortran/src/runtime -llfortran_runtime
duplicate symbol '__lcompilers_abs_f64_' in:
    a1.o
    a2.o
ld: 1 duplicate symbol for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

Hey, I'm getting similar errors while building Fitpack.

```console
/usr/bin/ld: fpclos.o: in function `_lcompilers_abs_f64_':
LFortran:(.text+0x0): multiple definition of `_lcompilers_abs_f64_'; fpbfout.o:LFortran:(.text+0x0): first defined here
/usr/bin/ld: fpcons.o: in function `_lcompilers_abs_f64_':
LFortran:(.text+0x0): multiple definition of `_lcompilers_abs_f64_'; fpbfout.o:LFortran:(.text+0x0): first defined here
/usr/bin/ld: fpcons.o: in function `_lcompilers_max0_i321_':
LFortran:(.text+0x40): multiple definition of `_lcompilers_max0_i321_'; concur.o:LFortran:(.text+0x0): first defined here
.
.
.
/usr/bin/ld: sphere.o: in function `_lcompilers_atan_f64_':
LFortran:(.text+0x0): multiple definition of `_lcompilers_atan_f64_'; fppola.o:LFortran:(.text+0x40): first defined here
/usr/bin/ld: surfit.o: in function `_lcompilers_max0_i321_':
LFortran:(.text+0x0): multiple definition of `_lcompilers_max0_i321_'; concur.o:LFortran:(.text+0x0): first defined here
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
I guess these will also be resolved with this?

>You may checkout to https://github.com/lfortran/lfortran/pull/2341, build lfortran and then use it to compile.

I did a checkout to this branch and on running the above files, I get:
```console
(lf) harshita@harshita-linux:~/Desktop/lfortran$ lfortran a.f90 -o a1.o -c --all-mangling --mangle-underscore --generate-object-code
(lf) harshita@harshita-linux:~/Desktop/lfortran$ lfortran b.f90 -o a2.o -c --rtlib --all-mangling --mangle-underscore --generate-object-code
(lf) harshita@harshita-linux:~/Desktop/lfortran$ clang -shared -fPIC -o lib.so a1.o a2.o -L/home/harshita/Desktop/lfortran/src/runtime -llfortran_runtime
/usr/bin/ld: a2.o: in function `_lcompilers_abs_f64_':
LFortran:(.text+0x0): multiple definition of `_lcompilers_abs_f64_'; a1.o:LFortran:(.text+0x0): first defined here
/usr/bin/ld: a2.o: in function `main':
LFortran:(.text+0x1020): multiple definition of `main'; a1.o:LFortran:(.text+0x60): first defined here
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

--- a/src/libasr/pass/unique_symbols.cpp
+++ b/src/libasr/pass/unique_symbols.cpp
@@ -58,14 +58,10 @@ class SymbolRenameVisitor: public ASR::BaseWalkVisitor<SymbolRenameVisitor> {
     std::string update_name(std::string curr_name) {
         if (startswith(curr_name, "_lpython") || startswith(curr_name, "_lfortran") ) {
             return curr_name;
-        } else if (startswith(curr_name, "_lcompilers_")) {
+        } else if (startswith(curr_name, "_lcompilers_") && current_scope) {
             // mangle intrinsic functions
-            if (current_scope) {
-                uint64_t hash = get_hash(current_scope->asr_owner);
-                return module_name + curr_name + "_" + std::to_string(hash) + "_" + lcompilers_unique_ID;
-            } else {
-                return module_name + curr_name + "_" + lcompilers_unique_ID;
-            }
+            uint64_t hash = get_hash(current_scope->asr_owner);
+            return module_name + curr_name + "_" + std::to_string(hash) + "_" + lcompilers_unique_ID;
         }
         return module_name + curr_name + "_" + lcompilers_unique_ID;
     }

--- a/src/libasr/pass/unique_symbols.cpp
+++ b/src/libasr/pass/unique_symbols.cpp
@@ -32,6 +32,11 @@ namespace LCompilers {
 
 using ASR::down_cast;
 
+uint64_t static inline get_hash(ASR::asr_t *node)
+{
+    return (uint64_t)node;
+}
+
 class SymbolRenameVisitor: public ASR::BaseWalkVisitor<SymbolRenameVisitor> {
     public:
     std::unordered_map<ASR::symbol_t*, std::string> sym_to_renamed;
@@ -42,6 +47,7 @@ class SymbolRenameVisitor: public ASR::BaseWalkVisitor<SymbolRenameVisitor> {
     bool bindc_mangling = false;
     bool should_mangle = false;
     std::string module_name = "";
+    SymbolTable* current_scope = nullptr;
 
     SymbolRenameVisitor(
     bool mm, bool gm, bool im, bool am, bool bcm) : module_name_mangling(mm),
@@ -52,12 +58,21 @@ class SymbolRenameVisitor: public ASR::BaseWalkVisitor<SymbolRenameVisitor> {
     std::string update_name(std::string curr_name) {
         if (startswith(curr_name, "_lpython") || startswith(curr_name, "_lfortran") ) {
             return curr_name;
+        } else if (startswith(curr_name, "_lcompilers_")) {
+            // mangle intrinsic functions
+            if (current_scope) {
+                uint64_t hash = get_hash(current_scope->asr_owner);
+                return module_name + curr_name + "_" + std::to_string(hash) + "_" + lcompilers_unique_ID;
+            } else {
+                return module_name + curr_name + "_" + lcompilers_unique_ID;
+            }
         }
         return module_name + curr_name + "_" + lcompilers_unique_ID;
     }
 
     void visit_TranslationUnit(const ASR::TranslationUnit_t &x) {
         ASR::TranslationUnit_t& xx = const_cast<ASR::TranslationUnit_t&>(x);
+        current_scope = xx.m_global_scope;
         std::unordered_map<ASR::symbol_t*, std::string> tmp_scope;
         for (auto &a : xx.m_global_scope->get_scope()) {
             visit_symbol(*a.second);

